src/Pandell.PackageRunner/Program.cs

```diff
@@ -81,20 +80,41 @@ private static int Main(string[] args)
         private int Run(Assembly packageRunnerAssembly, string packageRunnerExeFileName, string packageRunnerExeDirectory, string[] args)
         {
             var parameters = Program.ParseArguments(args);
+            var fileVersion = FileVersionInfo.GetVersionInfo(packageRunnerAssembly.Location).FileVersion;
+
+            if (parameters.ShowVersion || parameters.ShowHelp)
+            {
+                Console.WriteLine("PackageRunner  v" + fileVersion);
+            }
+
+            if (parameters.ShowHelp)
+            {
+                Program.ShowHelp();
+            }
+
+            if (parameters.ShowVersion || parameters.ShowHelp)
+            {
+                return 0;
+            }
 
             // Verify that assembly is signed and uses the correct key
+            var verbose = parameters.VerboseLog;
+            Action<string> writeVerboseMessage = message => Program.WriteDiagnosticsMessage(verbose ? TraceLevel.Verbose : TraceLevel.Off, message);
```

Comment:
My preference would be to call `Program.WriteDiagnosticsMessage` with the appropriate trace level each time instead of wrapping it in a delegate.